cypress/tests/ui/menu/menu.cy.js

```diff
@@ -1 +1,172 @@
 // All E2E Tests Related to menu goes here.
+
+import { prettifyName } from '../../../../src/utils';
+
+describe('Menu', () => {
+  it('verifies that users can select a section of the flowchart, through the drop down. #TC-16', function () {
+    // Alias
+    cy.intercept('GET', '/api/pipelines/*').as('pipelineRequest');
+    cy.get(':nth-child(2) > .menu-option__content > span').as('menuOption');
+
+    let menuOptionValue;
+
+    cy.get('@menuOption')
+      .invoke('text')
+      .then((menuOptionText) => {
+        menuOptionValue = menuOptionText;
+      });
+
+    // Action
+    cy.get('[data-test="kedro-pipeline-selector"]').click();
+    cy.get('@menuOption').click({ force: true });
+
+    // Adding an extra wait time due to flakiness
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(500);
+
+    // Assert after action
+    cy.wait('@pipelineRequest').then((interception) => {
+      // API Request URL
+      const requestUrl = interception.request.url;
+      expect(decodeURIComponent(requestUrl).toLowerCase()).to.include(
+        `/api/pipelines/${menuOptionValue.toLowerCase()}`
+      );
+
+      // Browser URL
+      cy.url().then((url) => {
+        const decodedURI = decodeURIComponent(url);
+        const queryParamValue = decodedURI.split('?')[1].split('=')[1];
+        expect(queryParamValue.toLowerCase()).to.be.eq(
+          menuOptionValue.toLowerCase()
+        );
+      });
+
+      // Pipeline Label in the Menu
+      cy.get('.pipeline-nodelist__row__label')
+        .first()
+        .invoke('text')
+        .then((pipelineLabel) => {
+          expect(menuOptionValue.toLowerCase()).to.include(
+            pipelineLabel.toLowerCase()
+          );
+        });
+    });
+  });
+
+  it('verifies that users can search/filter for a flowchart component using the search box. #TC-17', function () {
```

Comment:
this is cool, in future we have a ticket to enable both pretty/raw name search and then we can add further to this tst

Reply:
also sometimes there are multiple search results based on certain keywords like 'Feature'. do we want to include that? 
or maybe jest tests already cover it since we want to focus only on critical path.

Reply:
yes I have checked for a single instance. We can modify the test case in future to add other search functionality. We can include any keyword, I just chose Ingestion to filter out max elements